protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/StepContainer.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react'
+import { css } from 'styled-components'
+import { useSelector } from 'react-redux'
+import { createPortal } from 'react-dom'
+import {
+  ALIGN_CENTER,
+  BORDERS,
+  Box,
+  Btn,
+  COLORS,
+  Flex,
+  Icon,
+  JUSTIFY_CENTER,
+  JUSTIFY_SPACE_BETWEEN,
+  JUSTIFY_START,
+  OverflowBtn,
+  SPACING,
+  StyledText,
+} from '@opentrons/components'
+import { getUnsavedForm } from '../../../../step-forms/selectors'
+import { getTopPortalEl } from '../../../../components/portals/TopPortal'
+import { StepOverflowMenu } from './StepOverflowMenu'
+
+import type { IconName } from '@opentrons/components'
+
+const STARTING_DECK_STATE = 'Starting deck state'
+const FINAL_DECK_STATE = 'Final deck state'
+
+export interface StepContainerProps {
+  title: string
+  iconName: IconName
+  stepId?: string
+  iconColor?: string
+  onClick?: (event: React.MouseEvent) => void
+  onMouseEnter?: (event: React.MouseEvent) => void
+  onMouseLeave?: (event: React.MouseEvent) => void
+  selected?: boolean
+  hovered?: boolean
+}
+
+export function StepContainer(props: StepContainerProps): JSX.Element {
+  const {
+    stepId,
+    iconName,
+    onMouseEnter,
+    onMouseLeave,
+    selected,
+    onClick,
+    hovered,
+    iconColor,
+    title,
+  } = props
+  const formData = useSelector(getUnsavedForm)
+  const [top, setTop] = React.useState<number>(0)
+  const menuRoot = React.useRef<HTMLDivElement | null>(null)
+  const [stepOverflowMenu, setStepOverflowMenu] = React.useState<boolean>(false)
+  const isStartingOrEndingState =
+    title === STARTING_DECK_STATE || title === FINAL_DECK_STATE
+
+  let backgroundColor = isStartingOrEndingState ? COLORS.blue20 : COLORS.grey20
+  let color = COLORS.black90
+  if (selected) {
+    backgroundColor = COLORS.blue50
+    color = COLORS.white
+  }
+  if (hovered && !selected) {
+    backgroundColor = COLORS.grey30
+    color = COLORS.black90
+  }
+
+  const handleOverflowClick = (event: React.MouseEvent): void => {
+    const clickY = event.clientY
```

Comment:
```suggestion
    const { clientY } = event
```

and use `clientY` instead of `clickY`?